web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -120,13 +120,35 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         this.lois$,
         this.navigationService.getLocationOfInterestId$(),
       ]).subscribe(([survey, lois, locationOfInterestId]) => {
-        this.lois = lois;
-
-        this.onSurveyAndLocationsOfInterestUpdate(
-          survey,
-          lois,
-          locationOfInterestId
-        );
+        const loisMap = ImmutableMap(lois.map(loi => [loi.id, loi]));
+        const loiIdsToRemove = this.loisMap
```

Comment:
Can we do a since deep compare rather than checking ID and geometry separately here? 

If you populate a `Set` using a set of `Immutable.fromJS(loi)`, you can simply use `subtract()` on the two sets (before and after states) to get the added/modified and removed/modified values.

Reply:
That would be better, and that would require us to override `equals` function of `LocationOfInterest`. However, `LocationOfInterest` is still an abstract class with other inherent classes that we planned to remove (PointOfInterest, GeoJsonLocationOfInterest... in [loi.model.ts](https://github.com/google/ground-platform/blob/master/web/src/app/models/loi.model.ts)). So that's why I wanted to do the `Set.subtract()` thing later after we instantiate `LocationOfInterest` class and override `equals` function.

Filed #1444 & #1445 with descriptions and assigned to myself for now. Also updated the TODOs to relate to the bugs. WDYT?

Reply:
SGTM, thanks for filing those!